packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -112,6 +112,11 @@ class CreateCommand extends FlutterCommand {
       help: 'The organization responsible for your new Flutter project, in reverse domain name notation.\n'
             'This string is used in Java package names and as prefix in the iOS bundle identifier.'
     );
+    argParser.addOption(
+      'project',
+      defaultsTo: '<basename of the current folder>',
```

Comment:
`defaultsTo` isn't a help text: it's the actual default argument.  Your code avoids this invalid value only because it doesn't look at it if it wasn't supplied on the command line.  Better would be to have this default to null, and set it to the basename later if it's null.